src/node_blob.cc

```diff
@@ -379,6 +380,89 @@ void Blob::Reader::Pull(const FunctionCallbackInfo<Value>& args) {
       std::move(next), node::bob::OPTIONS_END, nullptr, 0));
 }
 
+void Blob::Reader::PullAll(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  Blob::Reader* reader;
+  ASSIGN_OR_RETURN_UNWRAP(&reader, args.Holder());
+
+  CHECK(args[0]->IsFunction());
+  Local<Function> fn = args[0].As<Function>();
+  CHECK(!fn->IsConstructor());
+
+  if (reader->eos_) {
+    Local<Value> arg = Int32::New(env->isolate(), bob::STATUS_EOS);
+    reader->MakeCallback(fn, 1, &arg);
+    return;
+  }
+
+  struct View {
+    std::shared_ptr<BackingStore> store;
+    size_t length;
+    size_t offset = 0;
+  };
+
+  struct Impl {
+    BaseObjectPtr<Blob::Reader> reader;
+    Global<Function> callback;
+    Environment* env;
+    std::function<void()> enqueue_cb;
+    std::vector<View> views;
+    size_t total_size = 0;
+  };
+
+  std::shared_ptr<Impl> impl_ptr = std::make_shared<Impl>();
+  impl_ptr->reader = BaseObjectPtr<Blob::Reader>(reader);
+  impl_ptr->callback.Reset(env->isolate(), fn);
+  impl_ptr->env = env;
+
+  auto next = [impl_ptr](int status, const DataQueue::Vec* vecs, size_t count, bob::Done doneCb) mutable {
+    Environment* env = impl_ptr->env;
+    HandleScope handleScope(env->isolate());
+    Local<Function> fn = impl_ptr->callback.Get(env->isolate());
+
+    if (status == bob::STATUS_EOS) impl_ptr->reader->eos_ = true;
+
+    if (count > 0) {
+      size_t total = 0;
+      for (size_t n = 0; n < count; n++) total += vecs[n].len;
+
+      std::shared_ptr<BackingStore> store = v8::ArrayBuffer::NewBackingStore(env->isolate(), total);
+      auto ptr = static_cast<uint8_t*>(store->Data());
+      for (size_t n = 0; n < count; n++) {
+        std::copy(vecs[n].base, vecs[n].base + vecs[n].len, ptr);
+        ptr += vecs[n].len;
+      }
+      std::move(doneCb)(0);
```

Comment:
I personnally do not understand why there is a std::move... If it is correct, I recommend adding a comment to explain it.